app/controllers/register-service.controller.js

```diff
@@ -129,22 +135,51 @@ module.exports = {
    * @param res
    * @returns {*|Promise|Promise.<T>}
    */
-  createPopulatedService: function createPopulatedService (req, res) {
+  createPopulatedService: async function createPopulatedService (req, res, next) {
+    const sessionData = req.register_invite
     const correlationId = req.correlationId
+    const code = req.register_invite.code
+    const otpCode = req.body['verify-code']
 
-    return registrationService.createPopulatedService(req.register_invite.code, correlationId)
-      .then(completeServiceInviteResponse => {
-        loginController.setupDirectLoginAfterRegister(req, res, completeServiceInviteResponse.user_external_id)
-        res.redirect(303, paths.selfCreateService.logUserIn)
-      })
-      .catch(err => {
-        if (err.errorCode === 409) {
-          const error = (err.message && err.message.errors) ? err.message.errors : 'Unable to process registration at this time'
-          renderErrorView(req, res, error, err.errorCode)
-        } else {
-          renderErrorView(req, res, 'Unable to process registration at this time', err.errorCode || 500)
+    const validOtp = validateOtp(otpCode)
+    if (!validOtp.valid) {
+      sessionData.recovered = {
+        errors: {
+          verificationCode: validOtp.message
         }
-      })
+      }
+      return res.redirect(303, paths.selfCreateService.otpVerify)
+    }
+
+    try {
+      await registrationService.submitServiceInviteOtpCode(code, otpCode, correlationId)
+    } catch (err) {
+      if (err.errorCode === 401) {
+        sessionData.recovered = {
+          errors: {
+            verificationCode: 'The verification code you’ve used is incorrect or has expired'
+          }
+        }
+        res.redirect(303, paths.selfCreateService.otpVerify)
+      } else if (err.errorCode === 410) {
+        return renderErrorView(req, res, 'This invitation is no longer valid', 410)
+      } else {
+        return renderErrorView(req, res, 'Unable to process registration at this time', err.errorCode || 500)
+      }
+    }
```

Comment:
The above code for validating the OTP code was previously in the `otp-verify.js` middleware, which was very confusing.